daemon/api_users.go

```diff
@@ -241,7 +244,26 @@ func postUsers(c *Command, r *http.Request, user *auth.UserState) Response {
 }
 
 func removeUser(c *Command, username string, opts postUserDeleteData) Response {
-	return NotImplemented("not implemented")
+	// catch silly errors
+	if username == "" {
+		return BadRequest("need a username to remove")
+	}
+	// first remove the system user
+	if err := osutilDelUser(username, &osutil.DelUserOptions{ExtraUsers: !release.OnClassic}); err != nil {
+		return InternalError(err.Error())
```

Comment:
Should we test this error condition (I think we don't right now)?

Reply:
you're right, I'll add more tests